### DISCUSSION TITLE
Welcome to jfa Discussions!

### DISCUSSION THREAD

**Discussion by koenderks** [koenderks/jfa#23] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Nasim-Afzali** [comment]
Hello,
I have some questions regarding stratified sampling using jfa audit package and need your advice:
1- How to calculate the sample size for each stratum depending on the type of allocation (proportional or optimal)?
2- Get a sample using stratified sampling, how to define which feature(s) as groups or "Strata" in Selecting function? I could not find the Strata parameter in Selecting function but Evaluating function has strata. 
3- How to get the Selecting function sample size of each stratum?
4- How do multi-stage stratified sampling or one-stage sampling at a deeper level of hierarchical allocation of the population (Nh)?
5- Can we have the sampling results of each stratum?
6- Is it possible to get the customized sample size of each stratum to the Selecting function?

Thank you in advance for your time and advice.

**Reply by koenderks** [reply]
Hi @Nasim-Afzali,

Currently, only the `evaluation()` function supports working with strata in the dataset. Stratified selection using `selection()` is in the works, and it will be in a later version of the **jfa** package. If you are not limited to using a frequentist or a Bayesian approach, my recommendation to you would be to simply select a random sample *n* items from each stratum and then check the resulting estimates using a Bayesian approach, as this approach is not limited you a pre-specified sampling plan.

So, for example, you would start by selecting 10 items from each stratum (let's assume 3 strata). Suppose you find 1 misstatement in the first stratum and 0 in the second and third strata, you would evaluate using:


```
set.seed(1)
x <- evaluation(x = c(1, 0, 0), n = c(10, 10, 10), method = "binomial", prior = TRUE)
summary(x)

	Bayesian Audit Sample Evaluation Summary

Options:
  Confidence level:               0.95 
  Method:                         binomial 
  Prior distribution:             Nonparametric 

Data:
  Sample size:                    30 
  Number of errors:               1 
  Sum of taints:                  1 

Results:
  Posterior distribution:         Nonparametric 
  Most likely error:              0.0923 
  95 percent credible interval:   [0, 0.20419] 
  Precision:                      0.11189 

Strata (3):
   n x t mle lb      ub precision
1 10 1 1 0.1  0 0.36436   0.26436
2 10 0 0 0.0  0 0.23840   0.23840
3 10 0 0 0.0  0 0.23840   0.23840
``` 

Then, if the data so far does not contain convincing evidence, you can simply sample more items from your population and evaluate in the same way (with `x` and `n` adjusted). E.g., I take 20 more observations in stratum 1 that do not contain an error:

```
set.seed(1)
x <- evaluation(x = c(1, 0, 0), n = c(30, 10, 10), method = "binomial", prior = TRUE)
summary(x)

	Bayesian Audit Sample Evaluation Summary

Options:
  Confidence level:               0.95 
  Method:                         binomial 
  Prior distribution:             Nonparametric 

Data:
  Sample size:                    50 
  Number of errors:               1 
  Sum of taints:                  1 

Results:
  Posterior distribution:         Nonparametric 
  Most likely error:              0.0492 
  95 percent credible interval:   [0, 0.15034] 
  Precision:                      0.10114 

Strata (3):
   n x t     mle lb      ub precision
1 30 1 1 0.03333  0 0.14409   0.11076
2 10 0 0 0.00000  0 0.23840   0.23840
3 10 0 0 0.00000  0 0.23840   0.23840
```

Hope this helps. If you have anymore questions, please let me know.

Short answers:
1. This is currently not implemented
2. This is currently not implemented
3. This is currently not implemented
4. This is currently not implemented
5. You can do this by calling the `summary()` function on the object returned by `evaluation()`.
6. This is currently not implemented
